climada/engine/supplychain.py

```diff
@@ -0,0 +1,322 @@
+        start_col, end_col = (4, 2468)
+
+        self.sectors = mriot.iloc[start_row:end_row_sectors, col_sectors].values
+        self.countries_iso3 = mriot.iloc[start_row:end_row, col_iso3].unique()
+        self.mriot_data = mriot.iloc[start_row:end_row,
+                                     start_col:end_col].values
+        self.total_prod = mriot.iloc[start_row:end_row, -1].values
+        self.cntry_pos = {
+            iso3: range(len(self.sectors)*i, len(self.sectors)*(i+1))
+            for i, iso3 in enumerate(self.countries_iso3)
+            }
+        self.mriot_type = 'wiod'
+
+    def calc_sector_direct_impact(self, hazard, exposure, imp_fun_set,
+                                  selected_subsec="service"):
+        """Calculate direct impacts.
+
+        Parameters:
+        ----------
+        hazard : Hazard
+            Hazard object for impact calculation.
+        exposure : Exposures
+            Exposures object for impact calculation.
+        imp_fun_set : ImpactFuncSet
+            Set of impact functions.
+        selected_subsec : str or list
+            Positions of the selected sectors. These positions can be either
+            defined by the user by passing a list of values, or by using built-in
+            sectors' aggregations passing a string with possible values being
+            "service", "manufacturing", "agriculture" or "mining". Default is "service".
+
+        """
+
+        if isinstance(selected_subsec, str):
+            built_in_subsec_pos = {'service': range(26, 56),
+                                   'manufacturing': range(4, 23),
+                                   'agriculture': range(0, 1),
+                                   'mining': range(3, 4)}
+            selected_subsec = built_in_subsec_pos[selected_subsec]
+
+        dates = [
+            dt.datetime.strptime(date, "%Y-%m-%d")
+            for date in hazard.get_event_date()
+            ]
+        self.years = np.unique([date.year for date in dates])
+
+        unique_regid_same_order = exposure.gdf.region_id.unique()
+        self.direct_impact = np.zeros(shape=(len(self.years),
+                                             len(self.countries_iso3)*len(self.sectors)))
+
+        self.cntry_dir_imp = []
+        for cntry in unique_regid_same_order:
+            cntyr_exp = Exposures(exposure.gdf[exposure.gdf.region_id == cntry])
+            cntyr_exp.check()
+
+            # Normalize exposure
+            total_ctry_value = cntyr_exp.gdf['value'].sum()
+            cntyr_exp.gdf['value'] /= total_ctry_value
+
+            # Calc impact for country
+            imp = Impact()
+            imp.calc(cntyr_exp, imp_fun_set, hazard)
+            imp_year_set = np.array(list(imp.calc_impact_year_set(imp).values()))
+
+            # Total production of country
+            cntry_iso3 = countries_by_numeric.get(str(cntry)).alpha3
+            idx_country = np.where(self.countries_iso3 == cntry_iso3)[0]
```

Comment:
Simplifying this would make it possible to use `admin1` region if one can supply the corresponding WIOT table.

Reply:
You need it as ROW is not an ISO code.

Reply:
I do not understand why this would be important? The only thing is you need to map the `reg_id` attribute to the WIOT attributes.
The `self.countries_iso3` is directly obtained from the WIOD table in the canonical use, and thus contains `ROW`. The cntry is obtained from the `red_ig` attribute of the exposures. Hence, it is the most natural thing to require the user to set the `reg_id` to `ROW` for the countries that belong to it right?

Reply:
There are about 40 countries in WIOD + ROW. The user should in principle be able to run a global analysis. Do we really want him to preprocess the exposures and 1) tranform the reg_id into iso3 codes and 2) filter the transformed reg_id to assess which ones are ROW and which ones can instead be found into WIOD? I do not think this is what we need.

Reply:
Good point about user friendliness! At the same time, hiding data structure assumptions deep in the code without documentation is not a good idea. Your code now assumes an exposure with country iso3 codes as `reg_id`. Not all exposures are necessarily this way, also not global ones. Worse, this is not even documented in the docstring! 

Hence, one possibility would be to write a separate helper function to convert an exposure with your assumed country iso3 code to the WIOD naming. 

The way it is now is not good because it assumes something about the exposures that is a) not necessary b) not documented c) restricts the code without reason.

Reply:
done in https://github.com/CLIMADA-project/climada_python/pull/180/commits/ed468290bf322d343c230549529e51c09832e9d0

Reply:
Amazing update!